src/main.rs

```diff
@@ -47,19 +54,21 @@ async fn main() {
     let interval = tokio::time::interval(Duration::from_secs(LOBBY_FLUSH_INTERVAL as u64));
     tokio::spawn(clear_lobby_on_interval(shared_state_clone, interval));
 
-    let oauth_client = oauth_client();
-
     let app = Router::new()
         .route("/hello_world", get(hello_world))
         .route("/auth/request_link", get(auth_client_string))
-        .route("/auth/authorised", get(authorised))
+        .route("/auth/callback/github", get(github_callback))
+        .route("/auth/callback/siwe", get(siwe_callback))
         .route("/slot/join", post(slot_join))
         .route("/contribute", post(contribute))
         .route("/info/status", get(status))
         .route("/info/jwt", get(jwt_info))
         .route("/info/current_state", get(current_state))
         .layer(Extension(shared_state))
-        .layer(Extension(oauth_client))
+        .layer(Extension(siwe_oauth_client()))
+        .layer(Extension(github_oauth_client()))
+        .layer(Extension(reqwest::Client::new()))
```

Comment:
For reqwest, we can make the http requests in the function that needs it, instead of adding this extension for it

Reply:
Going by the `reqwest` docs [here](https://docs.rs/reqwest/0.11.0/reqwest/struct.Client.html): 
> The Client holds a connection pool internally, so it is advised that you create one and reuse it.
So this is supposed to be more efficient :)